btk/catalog.py

```diff
@@ -53,8 +53,8 @@ def apply_selection_function(self, selection_function):
         if not callable(selection_function):
             raise TypeError("selection_function must be callable")
 
-        self.table = selection_function(self.table)
-        self._selection_functions.append(selection_function)
+        self.table = selection_function(self.table, *kwargs)
+        self._selection_functions.append((selection_function, *kwargs))
```

Comment:
The question remains here, if you pass keyword args then you need to save the dictionary in the tuple, and not unpack it =>
`self._selection_functions.append((selection_function, kwargs))`